watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/electrochemical_nutrient_removal/electrochemical_nutrient_removal.py

```diff
@@ -0,0 +1,223 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+import os
+from pyomo.environ import (
+    ConcreteModel,
+    Set,
+    Expression,
+    value,
+    TransformationFactory,
+    units as pyunits,
+    assert_optimal_termination,
+)
+from pyomo.network import Arc, SequentialDecomposition
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.util import get_solver
+import idaes.core.util.scaling as iscale
+from idaes.generic_models.unit_models import Product
+from idaes.generic_models.costing import UnitModelCostingBlock
+
+from watertap.core.wt_database import Database
+import watertap.core.zero_order_properties as prop_ZO
+from watertap.core.util.initialization import assert_degrees_of_freedom
+from watertap.unit_models.zero_order import (
+    FeedZO,
+    ElectroNPZO,
+)
+from watertap.core.zero_order_costing import ZeroOrderCosting
+
+
+def main():
+    m = build()
+    set_operating_conditions(m)
+    assert_degrees_of_freedom(m, 0)
+    assert_units_consistent(m)
+
+    solve(m)
+
+    display_results(m)
+
+    add_costing(m)
+
+    results = solve(m)
+    display_costing(m)
+    return m, results
+
+
+def build():
+    m = ConcreteModel()
+    m.db = Database()
+
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.prop = prop_ZO.WaterParameterBlock(
+        default={"solute_list": ["nitrogen", "phosphorus", "struvite"]}
+    )
+
+    # components
+    m.fs.feed = FeedZO(default={"property_package": m.fs.prop})
+    m.fs.product_H2O = Product(default={"property_package": m.fs.prop})
+    m.fs.product_struvite = Product(default={"property_package": m.fs.prop})
+    m.fs.electroNP = ElectroNPZO(
```

Comment:
@arao53 Do you recall if pumping costs are included within the electroNP unit? If not, maybe we should add a pump that connects to electroNP.

Reply:
Pumping costs are not included in the base unit - a pump electricity ZO unit has been added between the feed and electroNP to account for this.